zimbra/src/main/java/fr/openent/zimbra/worker/QueueWorker.java

```diff
@@ -29,6 +40,77 @@ public void start() throws Exception {
     public void startQueue() {
         this.running = true;
         this.workerStatus = QueueWorkerStatus.RUNNING;
+        while (this.running && !queue.isEmpty()) {
+            execute(this.queue.poll());
+        }
+    }
+
+    @Override
+    public void handle(Message<JsonObject> event) {
+        QueueWorkerAction action = QueueWorkerAction.valueOf(event.body().getString(Field.ACTION));
+        int maxQueueSize = event.body().getInteger(Field.MAXQUEUESIZE, configManager.getZimbraRecallWorkerMaxQueue());
+
+        switch (action) {
+            case REMOVE_TASK:
+                try {
+                    List<T> tasks = queueService.createTasksAndActionFromData(event.body().getJsonArray(Field.TASKS));
+                    removeTasks(tasks);
+                } catch (Exception e) {
+                    String errMessage = String.format("[Zimbra@%s::handle]:  " +
+                                    "an error has occurred while creating tasks: %s",
+                            this.getClass().getSimpleName(), e.getMessage());
+                    log.error(errMessage);
+                }
+            case SYNC_QUEUE:
+                this.syncQueue();
+                break;
+            case CLEAR_QUEUE:
+                this.clearQueue();
+                break;
+            case START:
+                this.startQueue();
+                break;
+            case PAUSE:
+                this.pauseQueue();
+                break;
+            case SET_MAX_QUEUE_SIZE:
+                this.setMaxQueueSize(maxQueueSize);
+                break;
+            case GET_STATUS:
+                this.sendWorkerStatus(event);
+                break;
+            case GET_REMAINING_SIZE:
+                this.sendRemainingSize(event);
+                break;
+            case UNKNOWN:
+            default:
+                log.error("[ZimbraConnector@%s::handle] Unknown QueueWorkerAction: %s", this.getClass().getSimpleName(), action.getValue());
+                break;
+        }
+    }
+    public void addTasks(List<T> tasks) {
+        if (tasks == null || tasks.isEmpty()) {
+            log.error("[ZimbraConnector@ICalRequestWorker:addTasks] taskIds cannot be null or empty");
+            return;
+        }
+        if (this.queue.size() + tasks.size() > this.maxQueueSize) {
+            log.warn("[ZimbraConnector@ICalRequestWorker:addTasks] Queue size limit is reached");
+        }
+
+        for(T taskIteration : tasks) {
+            if (!this.queue.contains(taskIteration) && (this.queue.size() < this.maxQueueSize)) {
```

Comment:
Potentiel problème de consistance sur ta collection `T` : https://rules.sonarsource.com/java/RSPEC-2175

Exemple de pb : https://community.sonarsource.com/t/java-s2175-using-number-as-key-type-in-a-map-produces-false-positives-when-int-is-used-when-accessing-the-map-value/32022

tu risques d'avoir un NPE